packages/uniforms/src/filterDOMProps.ts

```diff
@@ -1,15 +1,16 @@
 import { FilterDOMProps } from '.';
 
-type FilterDOMPropsKeys = keyof FilterDOMProps;
+/** @internal */
+export type FilterDOMPropsKeys = keyof FilterDOMProps;
 
 const registered: FilterDOMPropsKeys[] = [];
-const registeredCache = new Set<string>();
+const registeredCache = new Set<FilterDOMPropsKeys>();
 
 export const filterDOMProps = Object.assign(
   function filterDOMProps<T extends object>(props: T) {
     const filteredProps = { ...props };
     for (const prop in props) {
```

Comment:
Due to explicit typing of the cache, wouldn't this also make sense?

```suggestion
  function filterDOMProps<T extends FilterDOMPropsKeys>(props: T) {
    const filteredProps = { ...props };
    for (const prop in props) {
```

Reply:
No, as it'll require all of its properties to be required.